### ISSUE TITLE
Open issues DM7080

### ISSUE THREAD

**Issue by Hains** [Hains/dmm-bsp#7]
1)
RDEPENDS_kernel-image = "flash-scripts"
in dreambox-linux_3.4.bb

No recipe flash-scripts.

2)
I had to remove the line:
KERNEL_IMAGETYPE = "${@bb.utils.contains('MACHINE', 'dm520', 'vmlinux.gz', 'vmlinux.bin', d)}"

Also in dreambox-linux 3.4.bb

**Comment by persianpros** [comment]
About No.1 I guess this will do the job: https://github.com/Hains/dmm-bsp/pull/8/commits/29324e3757670d6e9eea4bbd2a9b5cc88db54c8a#diff-ce8fb5f41d834f5d0ecf018f445cb6e1

**Comment by Hains** [comment]
Yes indeed.I will do do that.

**Comment by persianpros** [comment]
Done my friend: https://github.com/Hains/dmm-bsp/pull/9 :)

**Comment by Hains** [comment]
Thanks.

**Comment by persianpros** [comment]
Check "flash-scripts" now please, I think "recovery" is providing it.
I did fix the missing "kernel-module-ath9k-htc' in https://github.com/Hains/dmm-bsp/pull/11 but "kernel-module-r8712u" were ok in models ...

**Comment by Hains** [comment]
I have license problem:
tpmd_1.7.bb: This recipe does not have the LICENSE field set (tpmd)

I removed tpmd it as workaround, but now error:

ERROR: Nothing RPROVIDES 'coreutils-realpath' (but /home/hains/test/openpli-oe-core/dmm-bsp/recipes-bsp/recovery/recovery_1.1.bb RDEPENDS

**Comment by persianpros** [comment]
That was my bad sorry, I did miss the "license-close.inc" file.
In some files we have this line:
LICENSE = "CLOSED"
If you see license warning about those files simply add this line just like what I did:
require conf/license/license-close.inc
See: https://github.com/Hains/dmm-bsp/pull/12/commits/14db3396dfac94867502f9d81383ccf991421507#diff-a910f6310b5f42818a0b2f5dc08854ec
You need "tpmd" for "tpm-ca" which is needed for "recovery".

**Comment by Hains** [comment]
Ok license problem solved. I restored tmpd.
But coreutils-realpath problem still remain.

**Comment by persianpros** [comment]
Maybe "coreutils" itslef isn't available in PLi's OE, Try adding https://github.com/openembedded/openembedded-core/blob/086d59455a7c273f7eccf442067df3c538f43c05/meta/recipes-core/coreutils/coreutils_8.25.bb and this folder https://github.com/openembedded/openembedded-core/tree/086d59455a7c273f7eccf442067df3c538f43c05/meta/recipes-core/coreutils/coreutils-8.25 to https://github.com/Hains/dmm-bsp/tree/master/recipes-core/coreutils If it solved the problem let me know to push the changes.

**Comment by Hains** [comment]
coreutils is built..But some slow/not responding gitbhub server at the moment. So it takes more time to test.

**Comment by Hains** [comment]
coreutils 8.25 is in openembedded-core openpli.. No need to  copy recipe and patches.They are equal BTW. The bbappend doesn't seem to work. no coreutils-realpatch..

**Comment by Hains** [comment]
Oh wait..i see a different between coreutils in openpli and the link you gave.
```
python __anonymous() {
	for prog in d.getVar('base_bindir_progs').split():
		d.setVarFlag('ALTERNATIVE_LINK_NAME', prog, '%s/%s' % (d.getVar('base_bindir'), prog))

	for prog in d.getVar('sbindir_progs').split():
		d.setVarFlag('ALTERNATIVE_LINK_NAME', prog, '%s/%s' % (d.getVar('sbindir'), prog))
}
```
in the link from you and
```
python __anonymous() {
	for prog in d.getVar('base_bindir_progs', True).split():
		d.setVarFlag('ALTERNATIVE_LINK_NAME', prog, '%s/%s' % (d.getVar('base_bindir', True), prog))

	for prog in d.getVar('sbindir_progs', True).split():
		d.setVarFlag('ALTERNATIVE_LINK_NAME', prog, '%s/%s' % (d.getVar('sbindir', True), prog))
}
```
in openembedded-core openpli..

BBCLASSEXTEND = "native nativesdk"

in

**Comment by Hains** [comment]
Your link with coreutils doesn't work.

Exception: TypeError: getVar() missing 1 required positional argument: 'expand'

coreutils in openembedded-core openpli is ok..but no coreutils-realpath.

**Comment by persianpros** [comment]
Try this on "coreutils_8.25.bbappend":
Change
RRECOMMENDS_${PN}_append_class-target = "${PN}-realpath ${PN}-stdbuf"
To
RRECOMMENDS_${PN} += "${PN}-realpath ${PN}-stdbuf"

**Comment by Hains** [comment]
Yes..bitbake -c cleanall coreutils and bitbake coreutils.  with your bbappend.
But still same problem.

**Comment by persianpros** [comment]
I think the problem about "bbappend" files is the "layer.conf": https://github.com/Hains/dmm-bsp/pull/14
My suggestion is create an organization via https://github.com/settings/organizations and call it "Dreambox-PLi" for example then move this repo there and add give me access to push the commits without the need of merge requests :)

**Comment by Hains** [comment]
Yes that's better. You don't have to wait for acceptance PR's anymore.
And i can do what i want to do it this repo and make a properly commit afterwards for DMM-PLI.

**Comment by persianpros** [comment]
Ok sounds good :) You can make your membership public here: https://github.com/orgs/DMM-PLi/people (As I did)
We're both owners with same access there :)
Also you can change the profile here: https://github.com/organizations/DMM-PLi/settings/profile

**Comment by Hains** [comment]
Ok thanks..

It's working BTW. but still license problem.
ERROR: tpmd-1.7-r0 do_populate_lic: QA Issue: tpmd: LIC_FILES_CHKSUM points to an invalid file: /home/hains/test/openpli-oe-core/build/tmp/work/mips32el-oe-linux/tpmd/1.7-r0/tpmd_1.7_mips32el/LICENSE-CLOSE [license-checksum]

But i can close issue 1 (solved).

**Comment by Hains** [comment]
I have that error also with e2openplugins when building feed. But i can ignore that error. It's a warning actually .I  think this error too.

**Comment by persianpros** [comment]
So can we ignore it? Or it stops the build?

**Comment by Hains** [comment]
Build is still running but It looks like we can ignore it.

**Comment by persianpros** [comment]
BTW I did add this commit for license: https://github.com/DMM-PLi/dmm-bsp/commit/1bf54f9b810faf625cc7a01f6af32f0ef318a94e

**Comment by persianpros** [comment]
As I don't own any kind of enigma2 STB it's really hard for me to test some things.
Even I don't have working build on my ubuntu and I'm working blind but soon I'll set up a build based on these sources and go deep ;)

**Comment by Hains** [comment]
Applied. It was already build so i don't see any warning anyway. But when build from stratch the warning should be dissapeared.
Restore kernel modules in enigma2-image.bb and no problem anymore.
Thanks.

I own a DM8000 only.. So i'm working blind too.
We need testers with new dreamboxes.

**Comment by persianpros** [comment]
I don't have any e2 box and not even a working build LOL.
My build server would be ready this week then I can help more :)

**Comment by Hains** [comment]
Not any e2..oh..ok.
I try another box later this week or next week.

**Comment by daimon680** [comment]
I am building for dm7080 is doing well

PREFERRED_VERSION_linux-dreambox = "3.4"  in openpli-oe-core  meta/dream   config/machine/include

**Comment by Hains** [comment]
openpli-oe-core/meta-dream/config/xxxx is for dm8000 only. So that must be "3.2". It's is not our layer.
When your BSP layer is up to date, dm7080 should build without any changes in meta-dream. I tested it myself.

**Comment by Hains** [comment]
Yes i see indeed your right sorry...

**Comment by Hains** [comment]
The problem is dreambox-mips32el.inc.. So we need dreambox-mips32el-new.inc

**Comment by Hains** [comment]
https://github.com/Hains/dmm-bsp/commit/ef62824eb2633ebceae8117d1abc3429f57910a5

Now it should work with any changes in meta-dream.

**Comment by Hains** [comment]
daimon680

Let us know if the image works? Thank you.

**Comment by daimon680** [comment]
build fail.

| ERROR: Function failed: do_install (log file is located at /home/daimon/7080/openpli-oe-core/build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.14/temp/log.do_install.20210)
ERROR: Task (/home/daimon/7080/openpli-oe-core/dmm-bsp/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install) failed with exit code '1'
NOTE: Tasks Summary: Attempted 2311 tasks of which 0 didn't need to be rerun and 1 failed.
NOTE: Writing buildhistory

Summary: 1 task failed:
  /home/daimon/7080/openpli-oe-core/dmm-bsp/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install
Summary: There were 6 WARNING messages shown.
Summary: There were 2 ERROR messages shown, returning a non-zero exit code.
make: *** [image] Errore 1

**Comment by Hains** [comment]
That isseu is still open. As workaround. Remove line 41
KERNEL_IMAGETYPE = "${@bb.utils.contains('MACHINE', 'dm520', 'vmlinux.gz', 'vmlinux.bin', d)}"

in openpli-oe-core/dmm-bsp/recipes-bsp/linux-dreambox_3.4.bb

**Comment by Hains** [comment]
The million dollar question is, does the box boot without it? Or is the present of vmlinux.gz sufficient?

**Comment by persianpros** [comment]
Best thing to do is build all models for once and check boot status later.

**Comment by daimon680** [comment]
build dm7080 finish ok.

**Comment by persianpros** [comment]
Please check https://github.com/DMM-PLi/dmm-bsp/commit/a1fac09d525b682dea4aa6882911a34379f8470c too, I did check it against https://github.com/opendreambox/opendreambox/blob/krogoth/meta-dreambox/recipes-kernel/linux/linux-dreambox_3.4.bb

**Comment by daimon680** [comment]
no boot

http://i64.tinypic.com/rvx7wk.png

**Comment by persianpros** [comment]
Maybe the reason is this: https://github.com/Hains/dmm-bsp/commit/e09310adc36bf1915c47f72655bc27748e9d8bf0

**Comment by daimon680** [comment]
Please check DMM-PLi/dmm-bsp@a1fac09 too, I did check it against 

ERROR: Task (/home/daimon/7080/openpli-oe-core/dmm-bsp/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install) failed with exit code '1'
NOTE: Tasks Summary: Attempted 5085 tasks of which 4288 didn't need to be rerun and 1 failed.
NOTE: Writing buildhistory

Summary: 1 task failed:
  /home/daimon/7080/openpli-oe-core/dmm-bsp/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install
Summary: There were 4 WARNING messages shown.
Summary: There was 1 ERROR message shown, returning a non-zero exit code.
make: *** [image] Errore 1

**Comment by Hains** [comment]
found something..

linux.bin is in directory:
build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.15/build/arch/mips/boot/vlinux.bin

When copy to:
build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.15/build/boot/vmlinux.bin

Then build ok and box probably boot.
Now figure out why vmlinux.bin is in wrong directory.

**Comment by Hains** [comment]
This does the trick
Line 24.

```
-  B = "${WORKDIR}/build"
+ B = "${WORKDIR}/build/arch/mips/boot/"
```

**Comment by daimon680** [comment]
vimlinux.bin present   in  build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.14/build/arch/mips/boot/vmlinux.bin

**Comment by Hains** [comment]
Can you copy that file to 
build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.15/build/boot/vmlinux.bin

Or better apply the path in my previous post.

**Comment by daimon680** [comment]
apply the path  
-  B = "${WORKDIR}/build"
+ B = "${WORKDIR}/build/arch/mips/boot/"

in openpli-oe-core/dmm-bsp/recipes-bsp/linux-dreambox_3.4.bb??

**Comment by Hains** [comment]
Yes 
Replace line 24 with
```
B = "${WORKDIR}/build/arch/mips/boot/"
```

**Comment by daimon680** [comment]
no boot
fatal  boot image too big
fatal failed  to run dreambox-secondstage .postinst

**Comment by daimon680** [comment]
Opening the tarball.xz of the image does not have the vmlinux.bin but is in gz format

**Comment by daimon680** [comment]
remove .gz in line   FILES_kernel-vmlinux = "/boot/vmlinux-${LINUX_VERSION}" in linux dreambox3.4.inc

add  KERNEL_IMAGETYPE = "${@bb.utils.contains('MACHINE', 'dm520', 'vmlinux.gz', 'vmlinux.bin', d)}" in linux-dreambox3.4.bb

build ok.

But always in the boot folder also  vmlinux.bin or vmlinux.gz???'

**Comment by Hains** [comment]
No no..that's old. forget it.
With up to date bsp layer(this one) it built fine.

But boot problem has something to do met secondstageloader.

Please be patience. This is still experimental.

**Comment by daimon680** [comment]
build fail kernel install
install: cannot stat './vmlinux.bin': No such file or directory
| WARNING: /home/daimon/7080/openpli-oe-core/build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.14/temp/run.do_install.13228:1 exit 1 from 'install -m 0644 ./${type} /home/daimon/7080/openpli-oe-core/build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.14/image/boot/${type}-3.4-4.0-dm7080'
| ERROR: Function failed: do_install (log file is located at /home/daimon/7080/openpli-oe-core/build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.14/temp/log.do_install.13228)
ERROR: Task (/home/daimon/7080/openpli-oe-core/meta-dream/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install) failed with exit code '1'
NOTE: Tasks Summary: Attempted 1487 tasks of which 1422 didn't need to be rerun and 1 failed.
NOTE: Writing buildhistory

Summary: 1 task failed:
  /home/daimon/7080/openpli-oe-core/meta-dream/recipes-bsp/linux/linux-dreambox_3.4.bb:do_install
Summary: There was 1 ERROR message shown, returning a non-zero exit code.
Makefile:127: set di istruzioni per l'obiettivo "image" non riuscito
make: *** [image] Errore 1

**Comment by daimon680** [comment]
in
https://github.com/DMM-PLi/meta-dream  fail build not kernel install
https://github.com/Hains/dmm-bsp   fail build not kernel install

**Comment by Hains** [comment]
Use DMM-PLi/meta-dream only. Forget my repo.
I will take a look..

**Comment by Hains** [comment]
As you can see we just started.. And a lot of changes...
In stead of adding submodule we now use git clone...
Did you run a 'make update' If so, your submodule is not up to date anymore.

1)
-Remove submodules dmm-bsp, also in makefile.
```
git submodule deinit -f dmm-bsp
```
-Make your build-env clean.No local changes.
-Follow instructions
https://github.com/DMM-PLi/meta-dream

**Comment by Hains** [comment]
Sorry, it's very chaotic at the moment.
Change github url, repo name, adding dm8000, git clone instead of submodule  etc....

**Comment by daimon680** [comment]
no problem

+	@echo 'export BB_ENV_EXTRAWHITE="MACHINE EXACTNAME"' > $@  error makefile

**Comment by daimon680** [comment]
new build https://github.com/DMM-PLi/meta-dream fail kernel install

**Comment by Hains** [comment]
vmlinux.bin again ?

I start build dm7080 now.

**Comment by daimon680** [comment]
Please check DMM-PLi/meta-dream@a1fac09 too, I did check it against

not work.

**Comment by Hains** [comment]
I have almost compile kernel..I think i know what's wrong.

**Comment by Hains** [comment]
Can you try in linux-dreambox.bb 
line 24
```
B = "${WORKDIR}/build"
```
and line 41
```
KERNEL_IMAGETYPE = "vmlinux"
```
I assume vmlinux will be included in image by:
KERNEL_EXTRA_OUTPUT ?= "arch/${ARCH}/boot/${KERNEL_EXTRA_IMAGETYPE}"
```
in linux-extra-image.inc

**Comment by Hains** [comment]
I give up.
The only way to make the build succesfull, is to rename vmlinux.bin to vmlinux.
So no vmlinux.bin.
I tried:
```
KERNEL_ALT_IMAGETYPE = "vmlinux.bin"
```
and more... and more....

**Comment by Hains** [comment]
https://github.com/Hains/dmm-bsp/commit/4dead44fc57cb722b23105ed2b955644fd2db52b

This will probably solve the --image too big---  error.
My mistake.
Image size if now decreased with +/- 6 MB.
And perhaps also secondstageloader problem due to image size ?

Prrr.. I have to beg, steal or borrow a dm7080hd, or of course to buy one.

**Comment by daimon680** [comment]
no vmlinux.bin in boot folder   in openpli-enigma2.homebuild.tar.xz
mmmm...  no boot.....

**Comment by Hains** [comment]
We need to copy that file. I see no other possibility.
From:
```
build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.15/build/arch/mips/boot/vmlinux.bin
```
to:
```
build/tmp/work/dm7080-oe-linux/linux-dreambox/3.4-r14.15/build/boot/vmlinux.bin
```
In recipe or manually for the time being.

**Comment by daimon680** [comment]
![schermata del 2017-07-13 20 22 20](https://user-images.githubusercontent.com/15824734/28181494-2901c240-6809-11e7-8fcb-d05ab2e19bbb.png)

my build openatv 6.0  in boot folder

**Comment by daimon680** [comment]
![schermata del 2017-07-13 20 26 05](https://user-images.githubusercontent.com/15824734/28181658-b4567c3c-6809-11e7-8b18-1bd57e270ef4.png)
 
my build openatv 6.1

**Comment by Hains** [comment]
We have the same files, plus backdrop.mvi, bootlogo.mvi and bootlogo_wait.mvi

But we have all vmlinux files.

New commit: restore KERNEL_IMAGETYPE = "vmlinux.bin" and copy. Well the build succeed at least.

https://github.com/Hains/dmm-bsp/commit/19db4b10b6dcbefd250caf32ae120ac6df7b14ca

BTW: vmlinux was present in /boot before this commit.

**Comment by daimon680** [comment]
no boot 
   
      [*] Remounting '/dev/mmcblk0p2' to '/data'
[*] Creating ext4 filesystem 'dreambox-rootfs' on /dev/mmcblk0p1
mke2fs 1.43-WIP (18-May-2015)
Creating filesystem with 243712 4k blocks and 60928 inodes
Filesystem UUID: 9a23ff0b-b630-4766-9a1d-fcac6604bb2b
Superblock backups stored on blocks: 
	32768, 98304, 163840, 229376

Allocating group tables: 0/8   done                            
Writing inode tables: 0/8   done                            
Creating journal (4096 blocks): done
Writing superblocks and filesystem accounting information: 0/8   done

[*] Mounting '/dev/mmcblk0p1' to '/mnt'
[*] Extracting '/tmp/DqRTSt' to '/mnt'
[*] Mounting '/dev' to '/mnt/dev'
[*] Mounting '/proc' to '/mnt/proc'
[*] Mounting '/run' to '/mnt/run'
[*] Mounting '/sys' to '/mnt/sys'
[*] Mounting '/tmp' to '/mnt/tmp'
/usr/sbin/librecovery: line 287: stat: command not found
/usr/sbin/librecovery: line 287: [: : integer expression expected
Fatal: Boot image is too big
Fatal: Failed to run dreambox-secondstage.postinst
[*] Unmounting '/mnt/tmp'
[*] Unmounting '/mnt/sys'
[*] Unmounting '/mnt/run'
[*] Unmounting '/mnt/proc'
[*] Unmounting '/mnt/dev'
[*] Unmounting '/mnt'

need help

**Comment by Hains** [comment]
Yes we are looking in the wrong direction. vmlinux.bin is not the problem.
The image does build now with latest changes due to vmlinux.bin, but still no boot.

You have in linux-dreambox-3.4.bb line 24
```
B = "${WORKDIR}/build"
```
?

I take a closer look to openATV build-environment these days.We are missing something in recipes.
But what?

**Comment by Hains** [comment]
Google on that error. They talk about:
-Rescue mode.
-Flash box back to original dream image and then to openpli or whatever.

Two years ago, i had same problem with dm8000. I flash the box with a openatv image.
So far so good. But when i went back to openpli, i had to flash to the box via serial(RS232) the 
secondstage-dm8000-84.nfi, before i could flash the box with OpenPLi image.

There are no exciting differents between OpenATV en this layer. except the do_install for vmlinux.bin to fix the build.
Unless OpenATV has some secret hidden files/code in their layer.

**Comment by daimon680** [comment]
https://dreambox.de/board/index.php?thread/21561-bootloader/

**Comment by daimon680** [comment]
look to   https://github.com/opendreambox/opendreambox  resolv problem boot

**Comment by Hains** [comment]
Thanks, now we getting somewhere.

Did you use meta-dream only, or the entire build environment?
I see a lot of differents. Less code in linux-dreambox-3.4.bb

**Comment by daimon680** [comment]
not solve
flash my build openatv 6.1 ok


[*] Mounting '/dev/mmcblk0p1' to '/mnt'
[*] Extracting '/tmp/jtE6z4' to '/mnt'
[*] Mounting '/dev' to '/mnt/dev'
[*] Mounting '/proc' to '/mnt/proc'
[*] Mounting '/run' to '/mnt/run'
[*] Mounting '/sys' to '/mnt/sys'
[*] Mounting '/tmp' to '/mnt/tmp'
[*] Writing /usr/share/dreambox-secondstage/ssbl.bin to /dev/mmcblk0 (LBA: 8192)
1035+1 records in
1035+1 records out
530316 bytes (517.9KB) copied, 1.391277 seconds, 372.2KB/s
[*] Creating boot image
[*] Writing bootblob.bin to /dev/mmcblk0 (LBA: 16384)
16801+0 records in
16801+0 records out
8602112 bytes (8.2MB) copied, 21.414661 seconds, 392.3KB/s
[*] Enabling boot source A
0+1 records in
0+1 records out
5 bytes (5B) copied, 0.001571 seconds, 3.1KB/s
[*] Unmounting '/mnt/dev'
[*] Unmounting '/mnt/proc'
[*] Unmounting '/mnt/run'
[*] Unmounting '/mnt/sys'
[*] Unmounting '/mnt/tmp'
[*] Unmounting '/mnt'
[*] Unmounting '/data'
[*] Finished successfully. You may reboot now.

**Comment by persianpros** [comment]
OK found the problem: https://github.com/openembedded/openembedded-core/commit/413a5faf99280f4e083358e4134aebbcfa47bb4c

PLi's OE isn't recent enough (tree: eb2fc2cd9081a4533ed30fe81c9f491b06cc5ae1 which is for Jan 12, 2017), OE-A and DMM are using newer versions of OE.

**Comment by Hains** [comment]
I hope OpenPLi will update openembedded-core soon. I have arch-linux users which build from my github repo for vuduo solo2 due to a too old openembedded-core.
Updating openembedded-core give a lot of build failures. Due to 'getvar' is deprecated in bitbake.
Enigma needs python-importlib. Introducing RSS(recipe specific sysyroots) and introducing HOSTTOOLS 
 seperate kernelmodules,changes in make image,.gcc 7 etc...

I have all patches..But for now, we have to wait...
